client/src/components/FormField.tsx

```diff
@@ -1,46 +1,47 @@
-import React from 'react';
-import Eye from '../assets/eye.svg';
-import NoShowEye from '../assets/eye-slash.svg';
-import classes from './FormField.module.scss';
-import { UseFormRegisterReturn } from 'react-hook-form';
+import { useState } from 'react';
+import { FieldError, UseFormRegisterReturn } from 'react-hook-form';
+import VisibilityOutlinedIcon from '@mui/icons-material/VisibilityOutlined';
+import VisibilityOffOutlinedIcon from '@mui/icons-material/VisibilityOffOutlined';
 
 type TextInputProps = {
   label: string;
   inputType?: string;
   placeHolder?: string;
   inputProps: UseFormRegisterReturn; // Should have the useForm register props
   inputClass?: string;
+  error?: FieldError;
 };
 
-export default function FormField(props: TextInputProps) {
-  const { label, inputType = 'text', inputProps, inputClass = '', placeHolder = label } = props;
-  const [shown, setShown] = React.useState(inputType !== 'password');
-  const toggleShown = () => {
-    setShown((prev) => !prev);
-  };
-  const eye = shown ? NoShowEye : Eye;
-  const effectiveInputType = inputType !== 'password' ? inputType : shown ? 'text' : 'password';
+export default function FormField({ label, inputType = 'text', inputProps, inputClass = '', placeHolder = label, error }: TextInputProps) {
+  const [showPassword, setShowPassword] = useState(false);
+  const effectiveInputType =
+    inputType !== 'password' ? inputType : showPassword ? 'text' : 'password';
+
   return (
-    <div className={classes['form-field']}>
-      <label htmlFor={`input-${inputProps.name}`}>{label}</label>
-      <div className={classes['form-field-input-container']}>
+    <div className="input-container">
+      <label htmlFor={`input-${inputProps.name}`} className='label'>{label}</label>
+      <div className="input-wrapper">
         <input
           {...inputProps!}
           id={`input-${inputProps.name}`}
+          dir="rtl"
           type={effectiveInputType}
-          className={`${classes['form-field-input']} ${inputClass}`}
+          aria-invalid={error ? 'true' : 'false'}
+          className={`form-input ${inputClass} ${error && 'invalid'}`}
           placeholder={placeHolder}
+          style={inputType === 'password' ? { paddingLeft: '2.75rem' } : {}}
         />
         {inputType === 'password' && (
-          <img
-            className={classes['form-field-eye']}
-            width={24}
-            height={24}
-            src={eye}
-            onClick={toggleShown}
-          />
+          <button type="button" onClick={() => setShowPassword((prev) => !prev)}>
+            {showPassword ? (
+              <VisibilityOffOutlinedIcon htmlColor="#304571" />
+            ) : (
+              <VisibilityOutlinedIcon htmlColor="#304571" />
+            )}
+          </button>
         )}
       </div>
+      {error && <p className='error' dir='rtl'>{error.message}</p>}
```

Comment:
I know it wasn't mentioned in the mockup, but the error messages are too close to the input labels.
You could maybe move it to the left or open a new issue

![image](https://github.com/RealByter/callforhelp/assets/58567842/a3fefe9c-0869-467b-8701-0be657ce0102)

Reply:
רשמתי את זה למעצבת, אני אנסה להזכיר לה. בעיקרון הודעות השגיאה לא אמורות להזיז את ה-layout כך שהבעיה היא עם המרחקים שיש עכשיו